extensions/ql-vscode/src/data-extensions-editor/data-extensions-editor-view.ts

```diff
@@ -86,15 +91,36 @@ export class DataExtensionsEditorView extends AbstractWebview<
         await this.postMessage({ t: "addModeledMethods", modeledMethods });
       },
     );
+    this.externalApiUsages = [];
   }
 
   public async openView() {
     const panel = await this.getPanel();
     panel.reveal(undefined, true);
 
+    panel.onDidChangeViewState(async () => {
+      if (await this.isTheMostRecentlyActivePanel()) {
+        await this.updateModelDetailsPanelState(
+          this.externalApiUsages,
+          this.databaseItem,
+        );
+      }
+    });
+
```

Comment:
Should we also add something that makes sure that any references to the panel are set to `undefined` when the panel is disposed? If we don't, you might expect methods on `mostRecentlyActivePanel` to work, while they would always fail if the panel is disposed.